external/js/kyber/lib/group/nist/point.js

```diff
@@ -0,0 +1,359 @@
+  }
+
+  // the point reference is stored in an object to make set()
+  // consistent.
+  this.ref = {
+    curve: curve,
+    point: curve.curve.point(_x, _y)
+  };
+}
+
+/**
+ * Returns the little endian representation of the y coordinate of
+ * the Point
+ *
+ * @return {string}
+ */
+Point.prototype.toString = function() {
+  if (this.ref.point.inf) {
+    return "(0,0)";
+  }
+  return (
+    "(" +
+    this.ref.point.x.fromRed().toString(10) +
+    "," +
+    this.ref.point.y.fromRed().toString(10) +
+    ")"
+  );
+};
+
+Point.prototype.string = Point.prototype.toString;
+Point.prototype.inspect = Point.prototype.toString;
+
+/**
+ * Tests for equality between two Points derived from the same group
+ *
+ * @param {module:group/nist~Point} p2 Point object to compare
+ * @return {boolean}
+ */
+Point.prototype.equal = function(p2) {
+  if (this.ref.point.isInfinity() ^ p2.ref.point.isInfinity()) {
+    return false;
+  }
+  if (this.ref.point.isInfinity() & p2.ref.point.isInfinity()) {
+    return true;
+  }
+  return (
+    this.ref.point.x.cmp(p2.ref.point.x) === 0 &&
+    this.ref.point.y.cmp(p2.ref.point.y) === 0
+  );
+};
+
+// Set point to be equal to p2
+
+/**
+ * set Set the current point to be equal to p2
+ *
+ * @param {module:group/nist~Point} p2 Point object
+ * @return {module:group/nist~Point}
+ */
+Point.prototype.set = function(p2) {
+  this.ref = p2.ref;
+  return this;
+};
+
+/**
+ * Creates a copy of the current point
+ *
+ * @return {module:group/nist~Point} new Point object
+ */
+Point.prototype.clone = function() {
+  const point = this.ref.point;
+  return new Point(this.ref.curve, point.x, point.y);
+};
+
+/**
+ * Set to the neutral element for the curve
+ * Modifies the receiver
+ *
+ * @return {module:group/nist~Point}
+ */
+Point.prototype.null = function() {
+  this.ref.point = this.ref.curve.curve.point(null, null);
+  return this;
+};
+
+/**
+ * Set to the standard base point for this curve
+ * Modifies the receiver
+ *
+ * @return {module:group/nist~Point}
+ */
+Point.prototype.base = function() {
+  const g = this.ref.curve.curve.g;
+  this.ref.point = this.ref.curve.curve.point(g.x, g.y);
+  return this;
+};
+
+/**
+ * Returns the length (in bytes) of the embedded data
+ *
+ * @return {number}
+ */
+Point.prototype.embedLen = function() {
+  // Reserve the most-significant 8 bits for pseudo-randomness.
+  // Reserve the least-significant 8 bits for embedded data length.
+  // (Hopefully it's unlikely we'll need >=2048-bit curves soon.)
+  return (this.ref.curve.curve.p.bitLength() - 8 - 8) >> 3;
+};
+
+/**
+ * Returns a Point with data embedded in the y coordinate
+ *
+ * @param {Uint8Array} data - data to embed with length <= embedLen
+ * @param {function} [callback] - to generate a random byte array of given length
+ *
+ * @throws {TypeError} if data is not Uint8Array
+ * @throws {Error} if data.length > embedLen
+ * @return {module:group/nist~Point}
+ */
+Point.prototype.embed = function(data, callback) {
```

Comment:
Is there no "stream" like things in Javascript when you just call stg like `stream.Next(32)` to get the next 32 bytes of output ? 
I guess it's not very Javascript-like, so at least is there somewhere where you provide a `random callback` using https://developer.mozilla.org/en-US/docs/Web/API/RandomSource/getRandomValues ? If not that'd be great to have, so developers don't have to always code the same callback.